ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```
@@ -7949,7 +7949,15 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
           }
         } else {
           if (tblDesc.isCTAS() && tblDesc.getStorageHandler() != null) {
-            tblDesc.setLocation(getCtasOrCMVLocation(tblDesc, viewDesc, createTableUseSuffix).toString());
+            try {
+              HiveStorageHandler storageHandler =
+                  HiveUtils.getStorageHandler(SessionState.getSessionConf(), tblDesc.getStorageHandler());
+              if (storageHandler.requiresLocationAtCreateTable()) {
+                tblDesc.setLocation(getCtasOrCMVLocation(tblDesc, viewDesc, false).toString());
+              }
+            } catch (Exception ex) {
+              LOG.warn("Error while creating HiveStorageHandler instance: {}", tblDesc.getStorageHandler(), ex);
```

is that OK, just to swallow the exception?